include/deal.II/dofs/dof_renumbering.h

```diff
@@ -689,6 +690,76 @@ namespace DoFRenumbering
                  const std::vector<unsigned int> &target_component =
                    std::vector<unsigned int>());
 
+
+
+  /**
+   * Convenience alias for variant type of supported FEValuesExtractors structs
+   */
+  using ExtractorVariant = std::variant<FEValuesExtractors::Scalar,
+                                        FEValuesExtractors::Vector,
+                                        FEValuesExtractors::Tensor<2>,
+                                        FEValuesExtractors::SymmetricTensor<2>>;
+
+
+  /**
+   * Utility function used to fill a given `component_order` vector by
+   * processing an `order` of FEValuesExtractors. Supported extractors are
+   * listed in DoFRenumbering::ExtractorVariant. The function is used in the
+   * implementation of component_wise(DoFHandler<dim, spacedim>&, const
+   * std::vector<ExtractorVariant> &).
+   */
+  template <int dim, int spacedim>
+  void
+  populate_component_order(
+    const std::vector<ExtractorVariant> &order,
+    std::vector<unsigned int>           &component_order,
+    unsigned int unassigned_value = numbers::invalid_unsigned_int);
```

Comment:
If it's a utility function, just move it into the `.cc` file as a function in an anonymous namespace without declaring it here. It should only be declared in the `.h` file if you think it's useful to users.